### PR TITLE
Handle validation of non expiring tokens in jwt_authn filter

### PR DESCRIPTION
**Pull request by vvinod1310**
For an explanation of how to fill out the fields, please see the relevant section 
in PULL_REQUESTS.md

*Description*: Handle validation of non expiring tokens in jwt_authn filter
*Risk Level*: Low
*Testing*: Added unit test to cover modification
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by ccaraman** [comment]
@vvinod1310 Please fix DCO https://github.com/envoyproxy/envoy/blob/dc03a9a4196204c13aaa192694acf67f3fc09ce8/CONTRIBUTING.md#dco-sign-your-work

And is there a test that can be added?

File: source/extensions/filters/http/jwt_authn/authenticator.cc [hunk 1]
```diff
@@ -119,7 +119,7 @@ void AuthenticatorImpl::verify(Http::HeaderMap& headers, Authenticator::Callback
   const auto unix_timestamp = std::chrono::duration_cast<std::chrono::seconds>(
                                   std::chrono::system_clock::now().time_since_epoch())
                                   .count();
-  if (jwt_.exp_ < unix_timestamp) {
+  if (jwt_.exp_ > 0 && jwt_.exp_ < unix_timestamp) {
```
Comment (lizan) [inline source/extensions/filters/http/jwt_authn/authenticator.cc]: @qiwzhang does jwt_verify_lib assign 0 to `exp_` when "exp" is not in the JWT?

Even if so, "0" is valid "exp", perhaps we can make `exp_` an `absl::optional<uint64_t>`?

Comment (vvinod1310) [reply source/extensions/filters/http/jwt_authn/authenticator.cc]: @ccaraman Fixed DCO and added a test thanks!
@lizan jwt_verify_lib assigns 0 to exp_ if not specified (Reference: https://github.com/google/jwt_verify_lib/blob/b06ba30f5d2049bcabd6b4ae8bb8f3518fb18b44/src/jwt.cc)

**Comment by vvinod1310** [comment]
Hi,

Could I request for this change to be reviewed and merge if appropriate?

Many thanks!

**Comment by mattklein123** [comment]
@lizan @qiwzhang PTAL

File: source/extensions/filters/http/jwt_authn/authenticator.cc (hunk 1 above)
Comment (lizan) [reply source/extensions/filters/http/jwt_authn/authenticator.cc]: OK thanks, can you add a comment to clarify the 0 case?

**Comment by vvinod1310** [comment]
@lizan Sure added comment please merge thanks!

File: include/envoy/json/json_object.h [hunk 2]
```diff
@@ -85,6 +85,18 @@ class Object {
    */
   virtual bool isNull() const PURE;
 
+  /**
```
Comment (lizan) [inline include/envoy/json/json_object.h]: these shouldn't appear in diff, can you merge master?

Comment (vvinod1310) [reply include/envoy/json/json_object.h]: @lizan done thanks!

**Comment by vvinod1310** [comment]
@lizan & @zuercher - thank you for approving!

**Comment by zuercher** [comment]
@vvinod1310 please update the description to include all the required fields from the pull request template.

**Comment by vvinod1310** [comment]
@zuercher  done thanks!

**Pull request merged**